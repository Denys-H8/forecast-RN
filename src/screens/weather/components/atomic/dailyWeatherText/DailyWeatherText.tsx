import React from 'react';
import { View } from 'react-native';

import CustomText from 'src/components/text';
import { styles } from './dailyWeatherText.styles';

interface IProps {
  text: string;
}

const DailyWeatherText: React.FC<IProps> = ({ text }) => {
  const words = text.split(' ');
  return (
    <View>
      {words.length > 1 ? (
        words.map((word) => (
          <CustomText key={word} style={styles.text}>
            {word}
          </CustomText>
        ))
      ) : (
        <CustomText style={styles.text}>{text}</CustomText>
      )}
    </View>
  );
};

export default DailyWeatherText;
