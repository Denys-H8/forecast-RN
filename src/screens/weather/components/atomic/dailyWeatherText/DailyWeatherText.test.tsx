import React from 'react';
import { render } from '@testing-library/react-native';
import DailyWeatherText from './DailyWeatherText';

const text = 'Some text';
const textLong = 'Ivano-Frankivsk';

describe('DailyWeatherText ', () => {
  test('should render DailyWeatherText', () => {
    const { toJSON } = render(<DailyWeatherText text={text} />);

    expect(toJSON()).toMatchSnapshot();
  });

  test('should render DailyWeatherText with long word', () => {
    const { toJSON } = render(<DailyWeatherText text={textLong} />);

    expect(toJSON()).toMatchSnapshot();
  });
});
