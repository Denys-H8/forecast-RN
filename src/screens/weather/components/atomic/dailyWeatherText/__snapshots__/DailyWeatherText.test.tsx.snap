// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DailyWeatherText  should render DailyWeatherText 1`] = `
<View>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "WorkSans-Regular",
        },
        Object {
          "color": "#8B8EA5",
          "fontSize": 12,
        },
      ]
    }
  >
    Some
  </Text>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "WorkSans-Regular",
        },
        Object {
          "color": "#8B8EA5",
          "fontSize": 12,
        },
      ]
    }
  >
    text
  </Text>
</View>
`;

exports[`DailyWeatherText  should render DailyWeatherText with long word 1`] = `
<View>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "WorkSans-Regular",
        },
        Object {
          "color": "#8B8EA5",
          "fontSize": 12,
        },
      ]
    }
  >
    Ivano-Frankivsk
  </Text>
</View>
`;
